api/controllers/zetter.ts

```diff
@@ -1,294 +1,328 @@
+        },
+    });
     res.status(201).json(newTweet);
 };
 
-export const updateTweet = (req: Request, res: Response) => {
-    const tweet: TweetType = req.body.tweet;
-    if (!tweet.numberOfFavorite) {
-        tweet.numberOfFavorite = 0;
+export const updateTweet = async (req: Request, res: Response) => {
+    const tweet: Tweet | null = await prisma.tweet.findUnique({
+        where: {
+            id: req.body.tweet.id,
+        },
+        include: {
+            user: true,
+            favorities: true,
+        },
+    });
+
+    if (!tweet) {
+        throw Error;
     }
 
     if (req.body.order === 'add') {
-        favorities.push({
-            id: favorities.length + 1,
-            // TODO: DBとつなぐときなおしたい
-            tweetId: tweet.id,
-            userId: req.user.id,
-            createdAt: new Date(),
+        await prisma.favorite.create({
+            data: {
+                tweetId: tweet.id,
+                userId: req.user.id,
+            },
+        });
+
+        await prisma.tweet.update({
+            where: {
+                id: tweet.id,
+            },
+            data: {
+                numberOfFavorite: tweet.numberOfFavorite + 1,
+            },
         });
-        tweet.isFavorite = true;
-        tweet.numberOfFavorite++;
     } else {
-        // favorities = favorities.filter(
-        //     (favorite) => favorite.tweetId !== req.body.tweet.id || favorite.userId !== req.user.id
-        // );
-        tweet.isFavorite = false;
-        tweet.numberOfFavorite--;
+        await prisma.favorite.delete({
+            where: {
+                tweetId_userId: {
+                    tweetId: tweet.id,
+                    userId: req.user.id,
+                },
+            },
+        });
+
+        await prisma.tweet.update({
+            where: {
+                id: tweet.id,
+            },
+            data: {
+                numberOfFavorite: tweet.numberOfFavorite - 1,
+            },
+        });
     }
     res.status(200).json(tweet);
 };
 
-export const getNotifications = (req: Request, res: Response) => {
+export const getNotifications = async (req: Request, res: Response) => {
     const limit: number = 10;
-    let notifications: Array<TweetType> = [];
-    let favoriteNotifications: Array<TweetType> = [];
 
-    const specificUsersTweets: Array<TweetType> = tweets.filter((tweet) => tweet.createdBy === req.user.id);
+    const specificUsersTweets = await prisma.tweet.findMany({
+        where: { createdBy: req.user.id },
+    });
+    const specificUsersTweetsIds = specificUsersTweets.map((obj) => obj.id);
 
     // リプライを取得
-    const replyNotifications: Array<TweetType> = tweets.filter((tweet) =>
-        specificUsersTweets.some((specificUsersTweet) => specificUsersTweet.id === tweet.replyTo)
-    );
+    const replyNotifications = await prisma.tweet.findMany({
+        where: { replyTo: { in: specificUsersTweetsIds } },
+        include: { user: true },
+    });
```

Comment:
https://github.com/sayanok/zetter/pull/39#discussion_r1012446796

schemaを↑なふうに書いた上で、
```suggestion
    const replyNotifications = await prisma.tweet.findMany({
        where: { replyTo: { createdBy: req.user.id} },
        include: { user: true },
    });
```
これで取れる